### PR TITLE
List option in group needs to register with Select

### PR DESCRIPTION
**Pull request by atmgrifter00**
# Pull Request

## 🤨 Rationale

Had a realization that list options that were in a group would not go through the `registerOption` routine (a fix for SLE Angular apps). While I'm not sure if there is any current use-case that would hit the problem, I went ahead and made the fix and the tests that demonstrate the problem.

## 👩‍💻 Implementation

Just made the private `getListOptionOwner` routine  on `ListOption` a bit more robust.

## 🧪 Testing

Wrote component and Angular tests modeled after existing tests but with a group as the option owner.

## ✅ Checklist

- [ ] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/list-option/index.ts [hunk 1]
```diff
@@ -66,9 +66,22 @@ export class ListOption extends FoundationListboxOption {
...
+
+        while (parent) {
+            if (this.isListOptionOwner(parent)) {
+                return parent;
+            }
+
+            parent = parent.parentElement;
         }
+
+        return null;
```
Comment (rajsite) [inline packages/nimble-components/src/list-option/index.ts:84]: prefer undefined over null

**Review by rajsite — APPROVED** [review]

File: packages/angular-workspace/nimble-angular/src/directives/select/tests/nimble-select-control-value-accessor.directive.spec.ts [hunk 2]
```diff
@@ -336,4 +337,158 @@ describe('Nimble select control value accessor', () => {
...
+                    </nimble-select>
+                </form>
+             `
+        })
+        class TestHostComponent implements OnChanges {
+            @ViewChild('select', { static: true }) public select: ElementRef<Select>;
+
+            @Input() public selectValue?: { name: string, value: number };
+
+            public selectOptions: { name: string, value: number }[] = [
```
Comment (rajsite) [inline packages/angular-workspace/nimble-angular/src/directives/select/tests/nimble-select-control-value-accessor.directive.spec.ts:432]: prefer as const

File: packages/nimble-components/src/select/tests/select.spec.ts [hunk 3]
```diff
@@ -500,6 +500,35 @@ describe('Select', () => {
...
+    it('option added directly to DOM under group synchronously registers with Select', async () => {
+        const { element, connect, disconnect } = await setupWithGroups();
+        await connect();
+        await waitForUpdatesAsync();
+        const newOption = new ListOption('foo', 'foo');
+        const registerOptionSpy = spyOn(
+            element,
+            'registerOption'
+        ).and.callThrough();
+        registerOptionSpy.calls.reset();
```
Comment (rajsite) [inline packages/nimble-components/src/select/tests/select.spec.ts:512]: do you need to reset a freshly made spy?

**Review by jattasNI — APPROVED** [review]

File: packages/angular-workspace/nimble-angular/src/directives/select/tests/nimble-select-control-value-accessor.directive.spec.ts (hunk 2 above)
Comment (atmgrifter00) [reply packages/angular-workspace/nimble-angular/src/directives/select/tests/nimble-select-control-value-accessor.directive.spec.ts]: I modify the array below, so it doesn't allow `as const`. I've removed the redundant typing declaration however.

File: packages/nimble-components/src/select/tests/select.spec.ts (hunk 3 above)
Comment (atmgrifter00) [reply packages/nimble-components/src/select/tests/select.spec.ts]: Nope.

**Pull request merged**